File: WPF/MVVM/TranslationConverter.vb
Imports System.Globalization
Imports System.Windows.Data

Namespace Common.MVVM

    Public Class TranslationConverter
        Implements IValueConverter

        Private Function GetLanguage() As KsLanguage
            Return KsApplication.Current?.Language
        End Function

        Public Function Convert(ByVal Value As Object, ByVal TargetType As Type, ByVal Parameter As Object, ByVal Culture As CultureInfo) As Object Implements IValueConverter.Convert
            Dim Lang = Me.GetLanguage()
            Dim Text = CType(Value, String)
            Return If(Lang?.Translation(Text), Text)
        End Function

        Public Function ConvertBack(ByVal Value As Object, ByVal TargetType As Type, ByVal Parameter As Object, ByVal Culture As CultureInfo) As Object Implements IValueConverter.ConvertBack
            Throw New NotSupportedException()
        End Function

    End Class

End Namespace

File: WPF/MVVM/TranslationConverter.cs
using System;
using System.Globalization;
using System.Windows.Data;

namespace Ks
{
    namespace Common.MVVM
    {
        public class TranslationConverter : IValueConverter
        {
            private KsLanguage GetLanguage()
            {
                return KsApplication.Current?.Language;
            }

            public object Convert(object Value, Type TargetType, object Parameter, CultureInfo Culture)
            {
                var Lang = this.GetLanguage();
                var Text = System.Convert.ToString(Value);
                return Lang?.GetTranslation(Text) ?? Text;
            }

            public object ConvertBack(object Value, Type TargetType, object Parameter, CultureInfo Culture)
            {
                throw new NotSupportedException();
            }
        }
    }
}
